src/compiler/utilities.ts

```diff
@@ -1538,6 +1538,34 @@ namespace ts {
         return getJSDocCommentsAndTags(node);
     }
 
+    export function getSourceOfAssignment(node: Node): Node {

```

Comment:
No need to export.

Please annotate potentially `undefined` return values (although as written this is potentially `false`).
I would prefer multiple statements with local variables instead of writing it all as one expression that has to be formatted on different lines anyway.
```ts
if (isExpressionStatement(node)) {
    const { expression } = node;
    if (isBinaryExpression(expression)) {
        const { operatorToken, right } = node;
        return operatorToken.kind === SyntaxKind.EqualsToken ? right : undefined;
    }
}
```